### PR TITLE
DBStructure is now using "q(" nevermore

### PR DESCRIPTION
**Pull request by annando**
The `q(...)` function is replaced with `DBA::` calls. Also the `information_schema` database is used whenever this was possible.

### CODE REVIEW

File: src/Database/DBStructure.php [hunk 1]
```diff
@@ -658,24 +656,24 @@ public static function update($basePath, $verbose, $action, $install = false, ar
 
 	private static function tableStructure($table)
 	{
-		$structures = q("DESCRIBE `%s`", $table);
-
-		$full_columns = q("SHOW FULL COLUMNS FROM `%s`", $table);
+		// This query doesn't seem to be executable as a prepared statement
+		$indexes = DBA::toArray(DBA::p(sprintf("SHOW INDEX FROM `%s`", $table)));
```
Comment (MrPetovan) [inline src/Database/DBStructure.php:660]: Please use `DBA::quoteIdentifier` instead of `sprintf()`.

File: src/Database/DBStructure.php [hunk 2]
```diff
@@ -708,35 +706,34 @@ private static function tableStructure($table)
...
-				$field["Type"] = str_replace($search, $replace, $field["Type"]);
+				$field['COLUMN_TYPE'] = str_replace($search, $replace, $field['COLUMN_TYPE']);
+
+				$fielddata[$field['COLUMN_NAME']]['type'] = $field['COLUMN_TYPE'];
 
-				$fielddata[$field["Field"]]["type"] = $field["Type"];
-				if ($field["Null"] == "NO") {
-					$fielddata[$field["Field"]]["not null"] = true;
+				if ($field['IS_NULLABLE'] == 'NO') {
+					$fielddata[$field['COLUMN_NAME']]['not null'] = 1;
```
Comment (MrPetovan) [inline src/Database/DBStructure.php:720]: Why the switch from boolean to integer?

Comment (annando) [reply src/Database/DBStructure.php:720]: By mistake? :-)

File: src/Database/DBStructure.php [hunk 3]
```diff
@@ -708,35 +706,34 @@ private static function tableStructure($table)
...
-				if ($field["Extra"] != "") {
-					$fielddata[$field["Field"]]["extra"] = $field["Extra"];
+				if (!empty($field['EXTRA'])) {
+					$fielddata[$field['COLUMN_NAME']]['extra'] = $field['EXTRA'];
 				}
 
-				if ($field["Key"] == "PRI") {
-					$fielddata[$field["Field"]]["primary"] = true;
+				if ($field['COLUMN_KEY'] == 'PRI') {
+					$fielddata[$field['COLUMN_NAME']]['primary'] = 1;
```
Comment (MrPetovan) [inline src/Database/DBStructure.php:732]: There's another type change here.

**Pull request merged**